### PR TITLE
Route1.0

### PR DESCRIPTION
**Pull request by miameng**
create a new variable route to replace the route list in the Server

### CODE REVIEW

File: Route.hs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+route =
```
Comment (david-yz-liu) [inline Route.hs]: you can delete this file

File: app/Route.hs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    import System.Log.Logger (updateGlobalLogger, rootLoggerName, setLevel, Priority(INFO))
+    import Data.String (fromString)
+    import FacebookUtilities
+    import Config (markdownPath, serverConf)
+    import qualified Data.Text.Lazy.IO as LazyIO
+    import Data.Int (Int64)
+
+
+route = 
+[ ("grid", gridResponse),
```
Comment (david-yz-liu) [inline app/Route.hs]: Put the open square bracket on the previous line, and please indent the tuples by 4 spaces.

File: app/Route.hs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  ("graph", graphResponse),
+  ("image", graphImageResponse),
+  ("timetable-image", timetableImageResponse x),
+  ("graph-fb",toResponse ""),
```
Comment (david-yz-liu) [inline app/Route.hs]: Check whitespace carefully - there should be no space before the comma, and a space after the comma.

File: app/Route.hs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  ("all-courses", allCourses),
+  ("graphs", queryGraphs),
+  ("course-info", courseInfo),
+  ( "depts", deptList),
+  ("timesearch",searchResponse),
+  ("calendar",calendarResponse),
+  ("get-json-data",getGraphJSON),
+  ("loading",loadingResponse),
+  ("save-json", saveGraphJSON),
+  (x, notFoundResponse)]
```
Comment (david-yz-liu) [inline app/Route.hs]: This one cannot be in here - the "not found" route will have to happen inside server, along with the `nullDir` and `seeOther` commands, which you have deleted (but should put back).

File: app/Server.hs [hunk 5]
```diff
@@ -23,6 +23,7 @@ import FacebookUtilities
 import Config (markdownPath, serverConf)
 import qualified Data.Text.Lazy.IO as LazyIO
 import Data.Int (Int64)
+import Route(route)
```
Comment (david-yz-liu) [inline app/Server.hs]: Put a space before the open parenthesis

File: app/Route.hs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    import Config (markdownPath, serverConf)
+    import qualified Data.Text.Lazy.IO as LazyIO
+    import Data.Int (Int64)
+
+
+route = 
+[ ("grid", gridResponse),
+  ("graph", graphResponse),
+  ("image", graphImageResponse),
+  ("timetable-image", timetableImageResponse x),
```
Comment (david-yz-liu) [inline app/Route.hs]: You can keep the `look` in here, at least for now. It is much harder to pull out the `look` as well, but that's not necessary.

**Comment by david-yz-liu** [comment]
@miameng you are making progress, which is good! You should have write an explicit type annotation for `route` (also, call it `routes`). It's a list of tuples, but for the type you need to state what the types are of the elements in each tuple (which is something you need to figure out by looking at `dir` and/or using the REPL).

You are trying to combine the routes using `msum` - that's not bad, but you aren't quite using it correctly. Keep in mind that you'll need to take the components of each tuple and call `dir` on them.

File: app/Server.hs [hunk 7]
```diff
@@ -23,6 +23,8 @@ import FacebookUtilities
 import Config (markdownPath, serverConf)
 import qualified Data.Text.Lazy.IO as LazyIO
 import Data.Int (Int64)
+<<<<<<< HEAD
```
Comment (david-yz-liu) [inline app/Server.hs]: This is a remnant of a merge conflict, and should be removed. (Always check for these before committing your work.)

File: app/Server.hs [hunk 8]
```diff
@@ -38,31 +40,9 @@ runServer = do
...
-              dir "depts" $ liftIO deptList,
-              dir "timesearch" searchResponse,
-              dir "calendar" $ lookCookieValue "selected-lectures" >>= calendarResponse,
-              dir "get-json-data" $ look "graphName" >>= \graphName -> liftIO $ getGraphJSON graphName,
-              dir "loading" $ look "size" >>= loadingResponse,
-              dir "save-json" $ look "jsonData" >>= \jsonStr -> look "nameData" >>= \nameStr -> liftIO $ saveGraphJSON jsonStr nameStr,
-              notFoundResponse
+              seeOther "graph" (toResponse "Redirecting to /graph"),    
+              map (dir . fst) routes,
+              not FoundResponse
```
Comment (david-yz-liu) [inline app/Server.hs]: This is a typo

**Comment by david-yz-liu** [comment]
@miameng to make the conversation today more explicit, you need to define a function `f :: (String, ServerPart Response) -> ServerPart Response` which you can use to map over your `routes` list. Work on defining such a function.

Also make sure to be compiling your program regularly by following instructions on the wiki. Finally, do **not** make any more changes directly on GitHub - you should be using a text editor to modify the files directly on your computer.

**Comment by miameng** [comment]
Hi David,

Thank you for these comments. As for today's meeting, I just want to let
you know that I did compile courseography at the very beginning of the
project, but it's been a while since last time I compiled it. So I forgot
about it.

I just compiled a few minutes ago. And actually for some routes that with
the 'look' function in them, I failed to pass in the string. Could you give
any advice on how to do that?

Thanks
Mia



*Mia Meng*



2016-06-30 19:23 GMT-04:00 David Liu <notifications@github.com>:

**Comment by david-yz-liu** [comment]
I'm not sure what error you are referring to, but regardless you should focus on fixing the other problems that were mentioned during the meeting.

File: app/Route.hs [hunk 9]
```diff
@@ -0,0 +1,47 @@
+module Route
+    (routes-
```
Comment (david-yz-liu) [inline app/Route.hs]: there seems to be a typo here

**Comment by david-yz-liu** [comment]
@miameng everything looks good, but these changes fail to compile due to the typo I mentioned. Please fix that as soon as possible so that I can merge in your work.

**Comment by david-yz-liu** [comment]
@miameng You did not fix the typo, but instead made other changes.

You need to make sure your program compiles on your machine before submitting any more changes to this pull request.

**Comment by miameng** [comment]
Hi David! I corrected the indentation of the square bracket, but it seems like there is another error which makes the code doesn't compile. The error message is "Process exited with code: ExitFailure 1." May I ask what that means?

**Comment by david-yz-liu** [comment]
That's not actually the error message, just a final status. The problems are the "not in scope" lines above that.

**Comment by david-yz-liu** [comment]
Thanks, @miameng!

**Pull request merged**